### PR TITLE
pmix/test: fixed tests

### PR DESCRIPTION
**Pull request by karasevb**
- Fixed spawn test
- Fixed fence test

### CODE REVIEW

File: test/pmix_test.c [hunk 1]
```diff
@@ -175,6 +178,8 @@ int main(int argc, char **argv)
         test_fail = 1;
     }
 
+    PMIX_WAIT_FOR_COMPLETION(spawn_wait);
```
Comment (artpol84) [inline test/pmix_test.c]: What if we are running non-spawn test?

File: test/pmix_test.c [hunk 2]
```diff
@@ -35,6 +35,9 @@
 
 #include "server_callbacks.h"
 #include "utils.h"
+#include "src/include/pmix_globals.h"
+
+int spawn_wait = 0;
```
Comment (artpol84) [inline test/pmix_test.c]: `spawn_wait` is `int` and set to 0 here

File: test/server_callbacks.c [hunk 3]
```diff
@@ -263,13 +266,27 @@ typedef struct {
...
+
+static void * _release_cb(void *arg)
 {
-    release_cbdata *cb = (release_cbdata*)cbdata;
+    release_cbdata *cb = (release_cbdata*)arg;
     if (NULL != cb->cbfunc) {
         cb->cbfunc(cb->status, "foobar", cb->cbdata);
     }
     free(cb);
+    spawn_wait = false;
```
Comment (artpol84) [inline test/server_callbacks.c]: but here `spawn_wait` is set to `false`. This will work but doesn't seem consistent.

**Comment by karasevb** [comment]
@artpol84 please check

**Pull request merged**